### ISSUE TITLE
Complete Rust AST structures

### ISSUE THREAD

**Issue by mwu-tow** [luna/luna#763]
### Summary
Complete the data structure presented at https://github.com/luna/enso/tree/wip/wd/rust-ast to correctly represent all Scala's parser outputs.

### Value
Will allow luna/enso#298 — producing AST in one language and using it in another.

### Specification
* most significant missing part are likely TextEscape structures, they should be possible to generate with macros
* Invalid constructor DanglingBase is also missing
* the AST also needs to be updated against the changes done in master in the meantime
* any discrepancies between ASTs in Scala and Rust should be removed.
* Scala AST might need adjustments as well, exact changes to be discussed with @wdanilo 

### Acceptance Criteria & Test Cases
Basically interested parties agree that it is good enough.
It will be validated and tested by the next task luna/enso#297.
